src/server/frontend_wayland/desktop_file_manager.cpp

```diff
@@ -55,6 +59,9 @@ std::string mf::DesktopFileManager::resolve_app_id(const scene::Surface* surface
     if (app)
         return app->id;
 
+    if (app_id_to_desktop_file_quirks.contains(app_id))
+        return app_id_to_desktop_file_quirks.at(app_id);
+
```

Comment:
Doing two lookups seems spuriously inefficient.

```suggestion
    for (auto const& q: app_id_to_desktop_file_quirks)
        if (q.first == app_id) return q.second;
```